### PR TITLE
ARROW-18120: [Release][Dev] Automate running binaries/wheels verifications

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by kou** [comment]
@raulcd What do you think about this?

File: dev/release/05-binary-upload.sh [hunk 1]
```diff
@@ -114,23 +116,37 @@ if [ ${UPLOAD_UBUNTU} -gt 0 ]; then
...
-    ARTIFACTORY_API_KEY="${ARTIFACTORY_API_KEY}" \
-    ARTIFACTS_DIR="${tmp_dir}/artifacts" \
-    DEB_PACKAGE_NAME=${DEB_PACKAGE_NAME:-} \
-    DRY_RUN=${DRY_RUN:-no} \
-    GPG_KEY_ID="${GPG_KEY_ID}" \
-    RC=${rc} \
-    STAGING=${STAGING:-no} \
-    VERSION=${version} \
-    YUM_TARGETS=$(IFS=,; echo "${yum_targets[*]}")
+if [ ${#rake_tasks[@]} -gt 1 ]; then
```
Comment (raulcd) [inline dev/release/05-binary-upload.sh:119]: whoah, I just learned what this does `${#rake_tasks[@]}` :exploding_head:

**Review by raulcd — COMMENTED** [review]
Thanks @kou ! The only think I am not sure is that this comment also triggers the `verify-rc-binaries-jars-*` but the jars are supposed to be uploaded on the `06-java-upload.sh`. I remember there was this issue that @kszucs opened https://issues.apache.org/jira/browse/ARROW-17260 that also might hide an issue around those

**Comment by kou** [comment]
> The only think I am not sure is that this comment also triggers the `verify-rc-binaries-jars-*` but the jars are supposed to be uploaded on the `06-java-upload.sh`.

Good catch!
Then, how about running `06-java-upload.sh` before `05-binary-upload.sh`? Or how about creating `07-binary-verify.sh`?

**Comment by kou** [comment]
> Or how about creating `07-binary-verify.sh`?

I chose this.

**Review by raulcd — APPROVED** [review]
Thanks @kou ! LGTM

**Pull request merged**